source/crud/tests/v2/updateWithPipelines.yml

```diff
@@ -64,6 +80,14 @@ tests:
         arguments:
             filter: { y: 1 }
             update: [ { $project: { x: 1 } }, { $addFields: { foo: 1 } } ]
+        result:
+          deletedCount: 0
+          insertedCount: 0
+          insertedIds: {}
+          matchedCount: 1
+          modifiedCount: 1
+          upsertedCount: 0
+          upsertedIds: {}
```

Comment:
The `result` for `findOneAndUpdate` will be the pre-modified document by default. In this case, the `result` should be one of the original data fixture documents instead of this UpdateResult. If you prefer to use the post-modified document for the `result` (and use the same value you'll have below in the outcome), you can add the `returnDocument: After` option to the `findOneAndUpdate` arguments to do so.

Since this also selects only one document, my previous comment about an ambiguous filter also applies here. I'd suggest selecting by `_id` or `x` to ensure we match only a single document.

Reply:
Added the `returnDocument` option and select `_id`.

Reply:
Unless I'm misreading the diff, the `result` assertion still needs to be changed to the returned document instead of an UpdateResult.

Reply:
I missed this. Since the result would be the same as the outcome, I have removed `result`. No need to check it twice.